cpu/stm32/periph/flashpage.c

```diff
@@ -229,13 +229,18 @@ void flashpage_write(void *target_addr, const void *data, size_t len)
     bool data_cache = FLASH->ACR & FLASH_ACR_DCEN;
     if (data_cache) {
         FLASH->ACR &= ~FLASH_ACR_DCEN;
+        /* Reset the data cache after it has been disabed */
+        FLASH->ACR |= FLASH_ACR_DCRST;
     }
 #endif
 #ifdef FLASH_ACR_ICEN
     /* Disable the instruction cache during page writes */
     bool instruction_cache = FLASH->ACR & FLASH_ACR_ICEN;
     if (instruction_cache) {
         FLASH->ACR &= ~FLASH_ACR_ICEN;
+        /* Reset the instruction cache after it has been disabed to avoid any
+           unintended operations later */
+        FLASH->ACR |= FLASH_ACR_ICRST;
```

Comment:
This should be wrapped in an ifdef only for `stm32wl`, the comment is not appropriate since it is not saying why we had to do this.

Reply:
Yes it is. Please check [6d1f754](https://github.com/RIOT-OS/RIOT/pull/16255/commits/6d1f75485179910c7e2df174d69cd82d9535edc4)